fixtures/xunit_tools.py

```diff
@@ -239,13 +251,21 @@ def pytest_collection_modifyitems(session, config, items):
     xml = etree.ElementTree(testcases)
     xml.write('test_case_import.xml', pretty_print=True)
 
+    test_name = []
     tests = []
     for item in items:
-        name = re.sub('\[.*\]', '', item.name)
-        try:
-            params = item.callspec.params
-            param_dict = {p: get_name(v) for p, v in params.iteritems()}
-        except:
-            param_dict = {}
+        if legacy:
+            if item.name in test_name:
+                continue
+            name = item.name
+            param_dict = None
+            test_name.append(name)
+        else:
+            name = re.sub(r'\[.*\]', '', item.name)
```

Comment:
@mfalesni I don't think it can, there are neither nested square brackets nor square brackets used in test name (other than for test parameter).